### PR TITLE
gh-137: deprecate `redshifts_from_nz` in favor of `redshifts`

### PR DESCRIPTION
**Pull request by Saransh-cpp**
I have started looking into tests for `galaxies.py` and this popped up during my research.

Closes: #137

### CODE REVIEW

**Review by paddyroddy — APPROVED** [review]
Good spot

File: glass/galaxies.py [hunk 1]
```diff
@@ -99,6 +103,13 @@ def redshifts_from_nz(
         samples from all populations.
 
     """
+    if warn:
+        warnings.warn(
+            "when sampling galaxies, redshifts_from_nz() is often not the function you"
+            " want. Try redshifts() instead. Use warn=False to suppress this warning.",
+            stacklevel=2,
+        )
```
Comment (Saransh-cpp) [inline glass/galaxies.py:111]: Maybe we don't need the `warn` keyword, and we should let the ignoring part be at the user end  (users can suppress warnings with something like `import warnings; warnings.filterwarnings("ignore")`)

Comment (ntessore) [reply glass/galaxies.py:111]: They can, with significantly more effort :)

Comment (ntessore) [reply glass/galaxies.py:111]: But, yeah, I agree that this is slightly muddying the separation of concerns. My reason for proposing the parameter is that I don't think many users would a priori know how to suppress a warning. So either we would have to let them figure it out by themselves, or explain it in the docs of the function, at which point the users would be blindly copying code. It seems easier to take the hit and add the flag ourselves.

**Review by paddyroddy — APPROVED** [review]
👌

**Pull request merged**